mage_ai/io/google_sheets.py

```diff
@@ -0,0 +1,289 @@
+            sheet_name (str, optional): A sheet name (title) to export the df. Defaults to None
+
+        Raises:
+            KeyError: _description_
+
+        Returns:
+            gspread.spreadsheet.Spreadsheet: A gspread Spreadsheet object.
+        """
+        sheet = None
+
+        if sheet_url:
+            sheet = self.client.open_by_url(sheet_url)
+        elif sheet_id:
+            sheet = self.client.open_by_key(sheet_id)
+        elif sheet_name:
+            sheet = self.client.open(sheet_name)
+        else:
+            raise KeyError(
+                'Please set one of sheet_url, sheet_id, or sheet_name to a valid Google Sheet.'
+            )
+        if sheet is not None:
+            with self.printer.print_msg(
+                f'Loaded Google Sheet \'{sheet.title}\' with ID \'{sheet.id}\''
+            ):
+                return sheet
+        else:
+            raise ValueError('There was an error loading the Google Sheet.')
+
+    def fetch_worksheet(
+        self,
+        sheet_url: str = None,
+        sheet_id: str = None,
+        sheet_name: str = None,
+        worksheet_name: str = None,
+        worksheet_position: int = 0,
+    ) -> gspread.worksheet.Worksheet:
+        """Fetches a worksheet from a Google Sheet, given a sheet_url, sheet_id, or sheet_name.
+        The worksheet may be fetched by name or position, but defaults to the first (0-indexed).
+
+        Args:
+            sheet_url (str, optional): A sheet URL to export the dataframe to. Defaults to None
+            sheet_id (str, optional): A sheet id to export the dataframe to. Defaults to None
+            sheet_name (str, optional): A sheet name (title) to export the df. Defaults to None
+            worksheet_name (str, optional): A worksheet name to export the df. Defaults to None
+            worksheet_position (int, optional): A worksheet position to export the df. Defaults to 0
+
+        Raises:
+            name_ex: Raised if a worksheet cannot be found with the specified name.
+            position_ex: Raised if a worksheet cannot be found at the specified position.
+            ValueError: Raised if there is an error loading the worksheet.
+
+        Returns:
+            gspread.worksheet.Worksheet: A worksheet object.
+        """
+        sheet = self.fetch_sheet(
+            sheet_url=sheet_url,
+            sheet_id=sheet_id,
+            sheet_name=sheet_name,
+        )
+
+        worksheet = None
+
+        if worksheet_name is not None:
+            worksheet = sheet.worksheet(worksheet_name)
+        else:
+            worksheet = sheet.get_worksheet(worksheet_position)
+
+        if worksheet is not None:
+            with self.printer.print_msg(
+                f'Loaded worksheet \'{worksheet.title}\' at index \'{worksheet.index}\''
+            ):
+                return worksheet
+        else:
+            raise ValueError('There was an error loading the worksheet.')
```

Comment:
I think you can just `return None` here. If the worksheet fails to load for some reason, I'm guessing the `gspread` code will throw an error.